pkg/imageverifier/image_verifier.go

```diff
@@ -0,0 +1,29 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package imageverifier
+
+import "context"
+
+type ImageVerifier interface {
+	Enabled(ctx context.Context) bool
```

Comment:
How is `Enabled` intended to be used here. There is already a built-in way to disable plugins in containerd and plugins can also disable themselves during Init.

Reply:
Similar thread here: https://github.com/containerd/containerd/pull/8493#discussion_r1188281709

These logs might be noisy if the user hasn't placed any verifier binaries, but would be useful for audit purposes:
```
INFO[2023-05-10T22:16:29.097720765Z] Verifying image pull                          digest="sha256:02bb6f428431fbc2809c5d1b41eab5a68350194fb508869a33cb1af4444c9b11" name="index.docker.io/library/alpine:latest#@sha256:c75ac27b49326926b803b9ed43bf088bc220d22556de1bc5f72d742c91398f69" verifier=bindir
INFO[2023-05-10T22:16:29.153495401Z] Image verifier allowed pull                   digest="sha256:02bb6f428431fbc2809c5d1b41eab5a68350194fb508869a33cb1af4444c9b11" name="index.docker.io/library/alpine:latest#@sha256:c75ac27b49326926b803b9ed43bf088bc220d22556de1bc5f72d742c91398f69" ok=true reason="accepted" verifier=bindir
```
The idea of was to avoid logging noise for the default image verifier configuration (i.e. no binaries in the verifier directory) by requiring opt-in via `Enabled` rather than opt-out via the `disabled_plugins` config field. If these logs aren't an issue, I can just remove `Enabled` from the interface. If the logs are an issue, we could remove `Enabled` and push the logs into the plugin implementation. Alternatively, users could just opt-out using the regular `disabled_plugins` mechanism if they don't like the logs.

Reply:
Can we move these logs to debug level?  I'd expect a denied pull to show up in logs, but not necessarily an allowed pull.